SoObjects/Mailer/SOGoMailBaseObject.m

```diff
@@ -206,6 +220,22 @@ - (NGImap4Connection *) imap4Connection
       [imap4 retain];
     }
 
+   // Connection broken, try to reconnect
+   if (![[[imap4 client] isConnected] boolValue])
```

Comment:
If we start the Logon with the IMAP server stopped, then this is failing with:
```
[SOGo: sogo_backend_list_contexts:343] - EXCEPTION: NSInvalidArgumentException, reason: Can not determine type information for -[NSNull (null)], backtrace:
```
I've added this:
```
-   if (![[[imap4 client] isConnected] boolValue])
+  if (![imap isKindOfClass: [NSNull class]]  && ![[[imap4 client] isConnected] boolValue])
```
to make it work. However, we have to check at OpenChange level the object is `nil` in `MAPIStoreUserContext`:
```
       connection = [currentFolder imap4Connection];
+      if (!connection)
+          [NSException raise: @"MAPIStoreIOException"
+                      format: @"Could not get a valid IMAP connection"];
+
```
However, when we start the IMAP server again, the connection is not restablished. The SOPE PR is applied for your information.